### ISSUE TITLE
java.io.IOException: error from app.callServiceExtension: "method not available: ext.flutter.inspector.getSelectedWidget"

### ISSUE THREAD

**Issue by tedhsieh1966** [flutter/flutter-intellij#2801]
## What happened

App crash every time after re-compiling my project after upgrading all the packages. Only work once the App was uninstalled from emulator, but crash again when re-start of the App.

## Version information

Android Studio `3.2.1` • Flutter plugin `io.flutter 29.1.1` • Dart plugin `181.5656`

Error getting Flutter sdk information.

## Exception

java.io.IOException: error from app.callServiceExtension: "method not available: ext.flutter.inspector.getSelectedWidget"

```
java.util.concurrent.CompletionException: java.io.IOException: error from app.callServiceExtension: "method not available: ext.flutter.inspector.getSelectedWidget"
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:593)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at io.flutter.run.daemon.DaemonApi$Command.completeExceptionally(DaemonApi.java:375)
	at io.flutter.run.daemon.DaemonApi.dispatch(DaemonApi.java:181)
	at io.flutter.run.daemon.DaemonApi$1.onTextAvailable(DaemonApi.java:141)
	at sun.reflect.GeneratedMethodAccessor59.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.execution.process.ProcessHandler$5.invoke(ProcessHandler.java:223)
	at com.sun.proxy.$Proxy17.onTextAvailable(Unknown Source)
	at com.intellij.execution.process.ProcessHandler.notifyTextAvailable(ProcessHandler.java:197)
	at com.intellij.execution.process.BaseOSProcessHandler$SimpleOutputReader.onTextAvailable(BaseOSProcessHandler.java:198)
	at com.intellij.util.io.BaseOutputReader.sendText(BaseOutputReader.java:202)
	at com.intellij.util.io.BaseOutputReader.processInput(BaseOutputReader.java:186)
	at com.intellij.util.io.BaseOutputReader.readAvailableNonBlocking(BaseOutputReader.java:105)
	at com.intellij.util.io.BaseDataReader.readAvailable(BaseDataReader.java:85)
	at com.intellij.util.io.BaseDataReader.doRun(BaseDataReader.java:163)
	at com.intellij.util.io.BaseDataReader$1$1.run(BaseDataReader.java:66)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:209)
	at com.intellij.util.io.BaseDataReader$1.run(BaseDataReader.java:63)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: error from app.callServiceExtension: "method not available: ext.flutter.inspector.getSelectedWidget"
	... 22 more
```
C:\flutter\bin\flutter.bat --no-color doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, v0.11.3-pre.12, on Microsoft Windows [Version 10.0.17134.345], locale zh-TW)
[√] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[√] Android Studio (version 3.2)
[√] Connected device (1 available)

• No issues found!
Process finished with exit code 0


C:\flutter\packages\flutter_tools\tool>C:\flutter\bin\flutter.bat --no-color doctor -v
[  +46 ms] executing: [C:\flutter\] git rev-parse --abbrev-ref --symbolic @{u}
[ +183 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[   +4 ms] origin/master
[   +1 ms] executing: [C:\flutter\] git rev-parse --abbrev-ref HEAD
[ +114 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] master
[   +1 ms] executing: [C:\flutter\] git ls-remote --get-url origin
[ +104 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] https://github.com/flutter/flutter.git
[   +1 ms] executing: [C:\flutter\] git log -n 1 --pretty=format:%H
[ +130 ms] Exit code 0 from: git log -n 1 --pretty=format:%H
[   +1 ms] c8ccd692ae6767a16ab0edd4061a4c59d24eaaaa
[   +1 ms] executing: [C:\flutter\] git log -n 1 --pretty=format:%ar
[ +126 ms] Exit code 0 from: git log -n 1 --pretty=format:%ar
[        ] 3 hours ago
[   +1 ms] executing: [C:\flutter\] git describe --match v*.*.* --first-parent --long --tags
[ +174 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v0.11.2-12-gc8ccd692ae
[+1765 ms] java -version
[ +190 ms] [√] Flutter (Channel master, v0.11.3-pre.12, on Microsoft Windows [Version 10.0.17134.345], locale zh-TW)
[   +5 ms]     • Flutter version 0.11.3-pre.12 at C:\flutter
[   +1 ms]     • Framework revision c8ccd692ae (3 hours ago), 2018-11-10 13:02:39 -0800
[   +1 ms]     • Engine revision 5646e86a6f
[        ]     • Dart version 2.1.0 (build 2.1.0-dev.9.3 9c07fb64c4)
-[   +5 ms] executing: c:\Android\Sdk\platform-tools\adb devices -l
[  +45 ms] Exit code 0 from: c:\Android\Sdk\platform-tools\adb devices -l
[        ] List of devices attached
           emulator-5554          device product:sdk_gphone_x86 model:Android_SDK_built_for_x86 device:generic_x86 transport_id:1
[  +23 ms] c:\Android\Sdk\platform-tools\adb -s emulator-5554 shell getprop                                                                                                                                                                     |
[ +190 ms] executing: c:\Android\Sdk\tools\bin\sdkmanager --licenses
[   +9 ms] ro.hardware = ranchu                                                                                                                                                                                                                 [
+2095 ms] [√] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[        ]     • Android SDK at c:\Android\Sdk
[        ]     • Android NDK location not configured (optional; useful for native profiling support)
[        ]     • Platform android-27, build-tools 27.0.3
[        ]     • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
[        ]     • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
[        ]     • All Android licenses accepted.
[   +1 ms] [√] Android Studio (version 3.2)
[        ]     • Android Studio at C:\Program Files\Android\Android Studio
[        ]     • Flutter plugin version 29.1.1
[        ]     • Dart plugin version 181.5656
[        ]     • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
[   +1 ms] [√] Connected device (1 available)
[        ]     • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)
[        ] • No issues found!
[  +16 ms] "flutter doctor" took 4,127ms.

**Comment by stevemessick** [comment]
@jacob314 Could you take a look at this? I'm wondering if there is some integration issue we need to fix.

**Comment by jacob314** [comment]
Can you run 
`flutter clean`
followed by 
`flutter pub get`
To see if the error reproduces?
This error is pretty surprising as that method has been in the flutter sdk for a long time.

**Comment by tedhsieh1966** [comment]
Thank you for the advice. Good news and Bad news. Yes, the problem was gone after flutter upgrade.
However, I continued to try simple_permission, and, BALA, the problem came back immediately, even I removed the code for simple_permission.
Now I recalled that the problem first appeared right after I tried simple_permission a while ago.
Below are the report.

I guess I need to remove flutter and reinstall it or wait for next upgrade to have it work again?

## Version information

Android Studio `3.2.1` • Flutter plugin `io.flutter 30.0.1` • Dart plugin `181.5656`

Error getting Flutter sdk information.

## Exception

java.util.concurrent.CompletionException: java.io.IOException: Application terminated

```
java.lang.Throwable: java.util.concurrent.CompletionException: java.io.IOException: Application terminated
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:126)
	at io.flutter.view.InspectorPanel.lambda$recomputeTreeRoot$5(InspectorPanel.java:575)
	at io.flutter.inspector.InspectorService$ObjectGroup.lambda$null$69(InspectorService.java:880)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:315)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:447)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:431)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:415)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:423)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by jacob314** [comment]
I've verified that simple_permission works fine with hot restart including when I asynchronously await a permission before calling main. If you have more details on exactly how you are using the library that would be very helpful.

There were two bug fixes that landed after you reported the bug that may have also fixed the issue.
We refined how we listened for service extensions to avoid using service extensions before they were available and we avoid spamming the user with spurious and confusing exceptions like the second one you reported.
